feat: dynamic protection, easy cert swap, auto save settings...

castdrian opened the pull request:
this pr aimes to resolve quite few more qol issues that were raised in the repo/discussed privately

- remove remaining unusernotification stuff
- ignore all cpp source and header generated warnings in xcode since they're part of subprojects
- resolve all deprecation warnings and all other remaining warnings (except uibutton contentinsets)
- implement dynamic ppqcheck protection setting which in turn resolves #141
- resolve #185 
- resolve #187 
- signing settings auto save now (will revert if you don't like, but found it quite annoying, frequently forgot to save and wondered why stuff isn't working)

khcrysalis commented:
> signing settings auto save now (will revert if you don't like, but found it quite annoying, frequently forgot to save and wondered why stuff isn't working)

when testing it doesn't seem to autosave when exiting out the view

castdrian commented:
> when testing it doesn't seem to autosave when exiting out the view

there was a teeny tiny mismatch with the switches and their settings, fixed and added some logging to prevent this from happening in the future
